crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -158,6 +161,18 @@ def upgrade_controller_nodes
         # FIXME: save the global status information that this node is being upgraded
         node_api.upgrade
 
+        # Explicitly mark node1 as the cluster founder
+        # this is one time action only (for each cluster)
+        Api::Pacemaker set_node_as_founder drbd_slave
+
+        # Remove "pre-upgrade" attribute from node1
+        # We must do it from the node where pacemaker is running
+        master_node_api = Api::Node.new drbd_master
+        master_node_api.remove_pre_upgrade_role_for drbd_slave
```

Comment:
Yeah, I'm not sure why you are thinking of doing that either. What are the advantages?

Also, shouldn't you check the return value of this call?